### PR TITLE
feat(ui): Replace fake NUMA data with API data.

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Removed all fake data from NUMA view and aggregate view.
- Populated NUMA view with API data. Next will be to populate the aggregate view using the same data.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the values listed on the NUMA view match what's given by the API. Note that master websocket API is currently sending fake NUMA data that has absurdly small hugepage values, so it doesn't appear on the donut chart.

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2084

### CODE REVIEW

**Comment by squidsoup** [comment]
Hey @Caleb-Ellis, a question about hugepage. My understanding of the data is that `allocated` and `free` represent the _number_ of hugepages, which have a size of 2K, so we should be multiplying `page_size` by `allocated` and `free` to determine the actual values to display.

image

File: ui/src/app/kvm/components/KVMResourcesCard/KVMResourcesCard.tsx [hunk 1]
```diff
@@ -12,32 +13,54 @@ import { MachineListTable } from "app/machines/views/MachineList/MachineListTabl
...
   };
   title?: string;
-  vfs?: ChartValues;
   vms: string[];
 };
 
+/**
+ * Returns a string with the formatted byte value and unit, e.g 1024 => "1KiB"
+ *
+ * @param {number} memory - the memory in bytes
```
Comment (squidsoup) [inline ui/src/app/kvm/components/KVMResourcesCard/KVMResourcesCard.tsx:34]: No need to add the types now, we can follow the [tsdoc](https://github.com/microsoft/tsdoc) standard which excludes types (as the code already documents these).

these can just read:

```
@param memory - the memory in bytes
```
etc.

**Review by squidsoup — APPROVED** [review]
Looks good @Caleb-Ellis, thanks! One note about js/tsdoc.

**Comment by squidsoup** [comment]
> Hey @Caleb-Ellis, a question about hugepage. My understanding of the data is that `allocated` and `free` represent the _number_ of hugepages, which have a size of 2K. Should we be multiplying `page_size` by `allocated` and `free` to determine the actual values to display?
> 
> image

discussed on irc, apparently this is just unrealistic fake hugepage data.

File: ui/src/app/kvm/components/KVMResourcesCard/KVMResourcesCard.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/kvm/components/KVMResourcesCard/KVMResourcesCard.tsx]: Got it, thanks!

**Pull request merged**